spec/features/clearance/visitor_signs_up_spec.rb

```diff
@@ -1,31 +1,43 @@
 # frozen_string_literal: true
 
-require "rails_helper"
-require "support/features/clearance_helpers"
+require 'rails_helper'
+require 'support/features/clearance_helpers'
 
-RSpec.feature "Visitor signs up" do
-  scenario "by navigating to the page" do
+RSpec.feature 'Visitor signs up' do
+  scenario 'by navigating to the page' do
     visit sign_in_path
 
-    click_link I18n.t("sessions.form.sign_up")
+    click_link I18n.t('sessions.form.sign_up')
 
     expect(current_path).to eq sign_up_path
   end
 
-  scenario "with valid email and password" do
-    sign_up_with "testname", "valid@example.com", "password"
+  scenario 'with valid email and password' do
+    sign_up_with 'testname', 'valid@example.com', 'password'
     visit sign_in_path
     expect_user_to_be_signed_in
   end
 
-  scenario "tries with invalid email" do
-    sign_up_with "testname", "invalid_email", "password"
+  scenario 'with valid email, password, and picture' do
+    user = FactoryBot.build(:user)
+    user.
```

Comment:
Should probably use `#attach_file`.

Reply:
:smile: Thanks for introducing me to `#attach_file`, @mike-burns. I read about it with [this Hashrocket post](https://til.hashrocket.com/posts/c790268652-attach-a-file-with-capybara) and the [rubydoc](https://www.rubydoc.info/github/jnicklas/capybara/Capybara%2FNode%2FActions%3Aattach_file). 

:thinking: Why would you suggest using that over `attach`? It has more simple arguments, but I am guessing the greater reason would be that it better reflects how a form would be used.

Adding: `attach_file("input-field-name", "public/apple-touch-icon.png")`  (and removing `user.picture.attach`) would allow capybara to test this more closely to the way a user would enter data on a page.

Any other reasons, or am I off base? TIA :)

Reply:
:file_folder: @orhanna14 What do you think of copying this image into a `spec/fixtures/` directory? I believe that using a resource in the `public/` dir makes the test more brittle as frontend resources could be swapped out or updated (renamed) at some point.

Reply:
Using `ActiveStore#attach` in the test is a test of ActiveStorage's `#attach` method; using Capybara's `#attach_file` method in the test is a test of the app. (I mentioned this more clearly [further down](https://github.com/orhanna14/homebrew-buddy/pull/21/files#r358995513), but `#attach_file` can be tricky to use.)

Reply:
I was able to utilize `attach_file` and refactor the code a bit to create a `sign_up_with_picture` method in clearance helpers as well. 🎉  I also moved the image to the `spec/fixtures` directory. Thanks for the valuable input!

Reply:
Oh heck yeah! I'm thrilled you got `attach_file` working!